Docs/articles/Kernel/VFS.md

```diff
@@ -72,122 +91,86 @@ foreach (var directoryEntry in directory_list)
-            Console.WriteLine("Content: ");
-            foreach (char ch in content)
-            {
-                Console.Write(ch.ToString());
-            }
-            Console.WriteLine();
-        }
+        var content = File.ReadAllText(file);
+
+        Console.WriteLine("File name: " + file);
+        Console.WriteLine("File size: " + content.Length);
+        Console.WriteLine("Content: " + content);
     }
 }
 catch(Exception e)
 {
     Console.WriteLine(e.ToString());
 }
 ```
-![Read File](https://raw.githubusercontent.com/CosmosOS/Cosmos/master/Docs/articles/Kernel/images/File%20System%20Read%20File.PNG)
 
-## Read specific file
+![Read File](https://raw.githubusercontent.com/CosmosOS/Cosmos/master/Docs/articles/Kernel/images/File%20System%20Read%20File.PNG)
 
-Now we will read specific file from a path.  
-As usual, we do it in a try catch block.
+## Create new file
+Reading and writing is working on existing files, but it's much more useful to write to our own files.
+Let's jump right into it:
 
 ```C#
 try
 {
-    var hello_file = fs.GetFile(@"0:\hello_from_elia.txt");
-    var hello_file_stream = hello_file.GetFileStream();
-
-    if (hello_file_stream.CanRead)
-    {
-        byte[] text_to_read = new byte[hello_file_stream.Length];
-        hello_file_stream.Read(text_to_read, 0, (int)hello_file_stream.Length);
-        Console.WriteLine(Encoding.Default.GetString(text_to_read));
-    }
+    var file_stream = File.Create(@"0:\testing.txt");
 }
 catch (Exception e)
 {
     Console.WriteLine(e.ToString());
 }
 ```
 
-We use the GetFile() method, which returns a file from a path.  
-Next we open a stream to the file and check if we can read it.  
-The last thing we do is reading from the stream, we use Read() method, which write the stream to a byte array.
+We can also [check our files list](https://github.com/CosmosOS/Cosmos/wiki/FAT-FileSystem#get-files-list) and see our new file in it.
 
-![Read Specific File](https://raw.githubusercontent.com/CosmosOS/Cosmos/master/Docs/articles/Kernel/images/File%20System%20Read%20Specified%20File.PNG)
-
-Notice that there is no much difference between this and [read all the files in a directory](https://github.com/CosmosOS/Cosmos/wiki/FAT-FileSystem#read-all-the-files-in-a-directory).  
-Actually the difference is by using GetFile() instead of GetDirectoryListing(), as the first returns single DirectoryEntry, and the second returns DirectoryEntryList.
+![Create File](https://raw.githubusercontent.com/CosmosOS/Cosmos/master/Docs/articles/Kernel/images/File%20System%20Create%20File.PNG)
 
-## Write to a file
+## Write to file
 
-Now we will write to an existing file.  
-Writing to a file is almost the same as reading from a file.  
-Always remember, we should use try catch block.  
+Now we will write to an existing file.
+Writing to a file is almost the same as reading from a file.
+Always remember that we should put our code in a try catch block.
```

Comment:
Could we eliminate some of these try {} catch (Exception ex) {} blocks? 
Any C# user should know that if a null parameter is passed to a method (in this case a non-existent file or directory), it's going to throw an exception.

If you just add a reminder about that at the beginning of the article that will suffice, In practice and for code cleanliness, I don't think it's a good idea to add a try-catch block every time a System.IO method is called. From personal experience, it's better to write project-specific FS functions that can catch these exceptions, as an operating system kernel will be highly frequently accessing IO. Maybe add a note for that too :+1:

Reply:
The problem is that currently the fs is not very stable and very easy to use incorrectly, which means if you dont use try catch blocks, errors will just cause the kernel to crash, making it a lot harder to debug. So I think its fine if we for now encourage users to use try catch blocks often.

Reply:
that's fair enough then :)  I'll go ahead and merge